[AOMP] Add shellcheck workflow

jplehr opened the pull request.

CatherineMoore committed:
Remove SimpleTimer.h due to invalid license.
Recode use of timer routines using <chrono> in tsp.cpp.

CatherineMoore committed:
1. Remove counting_iterator.h due to licensing issues
2. Update Makefile to use c++-20
3. Remove counting_iterator references in call to transform_reduce

jplehr commented:
This depends on https://github.com/ROCm/aomp/pull/1252 and makes only sense once that has landed.

dpalermo commented on .github/workflows/aomp-shell.yml:
```diff
@@ -0,0 +1,21 @@
+      - name: Set up shellcheck
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y shellcheck
```
I don't think we want to be running apt-get update & a possible install on every checkout of aomp.   I would drop out this part.  We can just install shellcheck as needed (I've added it to git/apps/openmp/sysadmin/install-3-other for new system setup/refresh)

gregrodgers replied:
I agree with Dan.  I added the install instructions to SOURCEINSTALL_PREREQUISITE.md

e-kwsm replied:
shellcheck is preinstalled.
https://github.com/actions/runner-images/blob/ubuntu24/20250511.1/images/ubuntu/Ubuntu2404-Readme.md#installed-apt-packages

dpalermo commented on .github/workflows/aomp-shell.yml:
```diff
@@ -0,0 +1,21 @@
+
+      - name: Run shellcheck
+        run: | 
+          find . -name "*.sh" | xargs shellcheck -S info {} \;
```
Also, do we really want to run shellcheck on all *.sh on every checkout?  Seems like it should be a commit hook instead on only the files being changed.

dpalermo replied:
Still not sure we want to run shellcheck on every script during the checkout.

jplehr replied:
not sure what you mean with `checkout` here?

The workflow will run on every push (to the AOMP repo) and when PRs against AOMP are opened / updated.
Note that when you work on your (user-) fork of AOMP, that action will run on the fork.

dpalermo replied:
When will this command run, and on what files?:
`find . -name "*.sh" | xargs shellcheck -S info {} \;`

jplehr replied:
This will run on Github runners when Github events happen:
- push to branch
- open/update PR.

It will then run the check on all .sh files in the repo and report the status back to the PR / repo.
No cycles on the user side are used.

dpalermo replied:
Running on all *.sh will mean this will produce errors until after all bash scripts in aomp will shellcheck clean...which is more than just waiting on https://github.com/ROCm/aomp/pull/1252.  Since it isn't blocking I guess that will be fine, users will need to sift through the output to find what they are actually changing.

jplehr replied:
Yeah, it's not optimal. If we want to put in more engineering to contain it to only the files touched in the PR or so, it can be done.
But I think this is OK to run w/o all updates already landed and incrementally work our way towards making this check green.

jplehr commented:
I think we want to wait for https://github.com/ROCm/aomp/pull/1463 to land and then use this wrapper instead of plain `shellcheck` in this workflow.

jtb20 commented:
As it is, the aomp-shellcheck script automatically modifies the files given as arguments in some circumstances (i.e. if they have failing checks in the "patchable" check list), which is probably not desirable for the GitHub workflow case.

jplehr commented:
Good point. Should there be a `--check-only` option or so that would point out issues and show how to use our wrapper to fix them?

jtb20 commented:
I think so, something like that.

jtb20 commented:
I've added a new --check-only option in the version up on https://github.com/ROCm/aomp/pull/1463 now.

e-kwsm commented on .github/workflows/aomp-shell.yml:
```diff
@@ -0,0 +1,16 @@
+
+      - name: Run shellcheck
+        run: | 
+          find . -name "*.sh" | xargs shellcheck -S info {} \;
```
The syntax is wrong, which must be
```suggestion
          find . -name "*.sh" -exec shellcheck -S info {} +
```
or
```suggestion
          find . -name "*.sh" | xargs shellcheck -S info
```

ampandey-1995 committed:
[SWDEV-483895][ASan]  Add 'hipMallocManaged' regression tests

jplehr committed:
[smoke] Adds valgrind `make` target (#1530)

This adds a new target `valgrind` to enable execution of a testcase
under valgrind control.

Only tested with a couple of simple examples: helloworld and some OMPT
test cases.

DominikAdamski committed:
[OpenMP] Update tests after upstream PR#154105 (#1532)

Use new values of no-loop and big-jump-loop mode constants

gregrodgers committed:
Initial support for building AOMP with TheRock

jplehr committed:
[smoke] Fixes potential nullptr deref (#1531)

When disabling OMPT support this test will crash with a segementation
fault.
With this fix the test will still fail but no longer seg fault.

gregrodgers committed:
turn off THEROCK_BUILD_TESTING because rccl-unittests fail. Also, change the logs to use the build directory

dhruvachak committed:
Move xteam-red-teams to smoke-dev before Xteam blocksize change. (#1533)

Lynd98 committed:
Change from rocprof to rocprofv3 because rocprof is depricated.

mhalk committed:
[smoke-dev][omptest] Move all omptest-based smoke-limbo tests

Reason: Avoid test-pipeline disturbance due to OpenMP move of omptest

Lynd98 committed:
Changed from rocprof to rocprofv3 because rocprof is depricated.

mhalk committed:
[smoke-dev][omptest] Fix omptest-based tests

Fix omptest-based + CMake tests
 - export of omptest_DIR fixes find_package(omptest)

Enhance build steps (header inclusion and linking) due to omptest improvements.

Test names:
 * omptest-device-emi
 * omptest-device-non-emi
 * veccopy-ompt-target-cmake

Lynd98 committed:
Changed rocprof to rocprofv3; rocprof is being depricated.

gregrodgers committed:
[tr_aomp] More updates.  See TODO file for completed and upcoming changes

dhruvachak committed:
Moved xteam-red-split-1 to smoke-dev before Xteam blocksize change. (#1540)

gregrodgers committed:
[tr_aomp] completed the two-step build 1)tr_clone_aomp.sh 2)tr_build_aomp.sh. The clone process was modified to ensure TheRock is on a shakey for the patch to fix issues

dhruvachak committed:
Updated tests after Xteam blocksize change from 1024 to 512. (#1545)

dpalermo committed:
[flang-rt] Add more control over hostdevice build/target/suffix (#1544)

Lynd98 committed:
Changed rocprof to rocprofv3 because rocprof is being depricated. (#1541)

Lynd98 committed:
Standardized how we call rocprofv3 in the test directory. (#1542)

mhalk committed:
[OpenMP][omptest] Adapt smoke-fails to omptest OpenMP move (#1543)

* [smoke] Fix omptest include directory

Fixed header include path (changed upstream)

* [smoke-fails] Fix omptest-based tests

Replace main() with corresponding macro
Reorder since GoogleTest executes in-order
Remove disabled test as this causes a "fail"
Split certain suites into SuitePass and SuiteFail
 - Reason: TEST and TEST_XFAIL represent different fixture classes
           But only one fixture class per suite is allowed

skatrak committed:
[smoke-fort] Move passing test to smoke-fort (#1539)

gregrodgers committed:
[tr_aomp] Remove use of manifest file and implement policy where we freeze at a specified hash hey for TheRock and its specific submodules

gregrodgers committed:
[tr_aomp] update TheRock AOMP README and TODO

dpalermo committed:
[smoke-fort-fails] Add flang-535320,535416*,537499 (#1546)

dpalermo committed:
[smoke-fort-dev] flang-471469: compute HIP_LIB_PATH from AOMP (#1547)

* [smoke-fort-dev] flang-471469: compute HIP_LIB_PATH from AOMP

* Add ?=

jplehr committed:
[CK] Fix spelling of CMake option (#1548)

gregrodgers committed:
[tr_aomp] created build_install_aomp_from_therock.sh to use for incremental compiler updates.  Added convenience links for llvm-project/build. Updated README and TODO file

mhalk committed:
[OpenMP][omptest] Adopt omptest installation guard into scripts

Enable installation of omptest for 'project' and 'openmp' components by default.
See: https://github.com/llvm/llvm-project/pull/155433

dpalermo committed:
[smoke-fort-dev] hipfort: compute HIP_LIB_PATH from AOMP (#1550)

gregrodgers committed:
[tr_aomp] Removed hash file and put TheRock shakey in a new info file associated with the AOMP release.

gregrodgers committed:
[tr_aomp] update README with speedup advice

gregrodgers committed:
[tr_aomp] Change the convenience link llvm-project to be soft relative.  This avoids creating a recursive link

jplehr committed:
[NFC] Remove trailing blanks (#1551)

gregrodgers committed:
[tr_aomp] Removed need for TheRock.orig and reuse TheRock when it exists to significantly speed up tr_clone_aomp.sh.  The rarely used create_patch tool now builds an orig.TheRock from saved branches in each submodule

gregrodgers committed:
stop building rccl and math libs till we have the compiler working nicely

gregrodgers committed:
[tr_aomp] Missing info file will start a new release. Also removed logging to side files during build and let message go to the screen

agozillon committed:
[Flang][Test] Add new test-case to use_dev_addr_fort.f90

Adding a new test case I found doesn't work in amd-staging at the moment, but should soon when
I land a new PR. But effectively we crash when we have a map and use_device_addr/ptr to the same
list item on a target at the moment. New patch should allow this.

gregrodgers committed:
[tr_aomp] Moved all release files to a release directory, fixed clone to only run fetch_sources one time. It also only appies the set of patches one time.  Instead of a single patch file requiring a copy of the repo, we now just use git diff on the TheRock and eachof the submodules.

gregrodgers committed:
add patchelf to ubuntu prereq

dhruvachak committed:
Added No-Loop test with over-subscription options. (#1553)

gregrodgers committed:
[tr_aomp] fixed logging messages and set THEROCK_AMDGPU_TARGETS to include all GPU isn GFXLIST, just like aomp

jplehr committed:
[smoke-dev] Link Makefile.rules instead of file (#1554)

In all other instances the Makefile.rules is a symlink to the
smoke/Makefile.rules file. This patch changes this likely accident to
also have smoke-dev link to that file.

jplehr committed:
[Fix] Set the symlink correctly (#1556)

I introduced the wrong symlink in https://github.com/ROCm/aomp/pull/1554

jplehr committed:
[Fix] Have RUNENV as env not as executable in valgrind (#1555)

With the previous location valgrind would complain that it cannot find
the executable.

CatherineMoore committed:
Add AMD Copyright and MIT license designation.

dpalermo committed:
[tr_aomp] Fixup .amd-llvm.smrev to use latest llvm-project SHA (#1560)

jplehr committed:
[Bbots] Add libc build dependency (#1562)

Libc headergen requires pyyaml to be available.

dpalermo committed:
[tr_aomp] Update GFXLIST to current list supported by TheRock (#1561)

dpalermo committed:
[tr_aomp] Change default repo & install from tr_aomp to tr-aomp (#1564)

- underscore is otherwise used as the separator between the
     name and the version (e.g. tr-aomp_22.0-5)

jplehr committed:
[BBots] Add missing deps for offload unit tests (#1565)

It seems that the offload unit tests did complain about some headers
missing when running on the bare ubuntu image.
This adds the missing deps.

estewart08 committed:
[build] - Additional cmake variables added to support devicertl move (#1563)

DominikAdamski committed:
[OpenMP][Clang] Add test for nested parallel for (#1566)

gregrodgers committed:
[tr_aomp] Add two new skip variables and update README

dpalermo committed:
[tr_aomp] Generate .amd-llvm.smrev with latest llvm-project SHA (#1568)

dpalermo committed:
[tr_aomp] Move GFXLIST to tr_aomp_common_vars (#1569)

* [tr_aomp] Move GFXLIST to tr_aomp_common_vars

    - Allows sourcing before running other scripts
    - Needed before running build_llvm-flang-rt-host-dev.sh

* Remove trailing blank line

gregrodgers committed:
[tr-aomp] Big cleanup of tr_build_aomp.sh and move to aomp_22.0-6 with tip of TheRock from today

gregrodgers committed:
[tr-aomp] deleted junk files that got in last commit

dpalermo committed:
[aomp/trunk] Add amdgcn cmake flags to build_project.sh (#1571)

gregrodgers committed:
[tr_aomp] Copy FileCheck binary into lib/llvm/bin directory for smoke testing

gregrodgers committed:
[tr_aomp] No longer plan to use this script.  Instead, we will use ./tr_build_aomp.sh restart

gregrodgers committed:
[tr_aomp] update README to describe the use of the restart option for tr_build_aomp.sh.  Also change the defaults to not skip components.  Users will have to override these SKIP vars to 1 to shorten initial build times

gregrodgers committed:
[tr_aomp] turn of build of ck because it is currently failing

dhruvachak committed:
Added test for occupancy-based num_teams for generic spmd kernels. (#1573)

Modified existing tests based on occupancy.

gregrodgers committed:
[tr_aomp] Improve the reset of TheRock repo when switching hash keys for a new AOMP release

gregrodgers committed:
[tr_aomp] enable clang-tidy and remove diagnostic cmake message

estewart08 committed:
[build] - Updates for using ROCm 7.0 sources (#1574)

dhruvachak committed:
Fixed smoke-dev reduction failures: use fabs from math library, not abs from stdlib. (#1575)

Calls from stdlib may not be supported in target regions. DeviceRTL earlier resolved
some of these calls but that cannot be relied on. That behavior has changed with the
move of Device RTL.

gregrodgers committed:
[rocprofiler-register] turn off build for TESTS because hip is not yet available when rocprofiler-register is built

estewart08 committed:
[build] - Disable building tests for rocprofiler-sdk (#1577)

A recent commit came into the 7.0 branch which seems
to be dependent on another commit and is causing a build
failure. For now we can just disable the tests to avoid
the issue.

agozillon committed:
Fix UMT build by rebasing and tweaking the currently required patch for 5.9.0 (#1578)

dpalermo committed:
[UMT] Add more error checking to builds/tests (#1579)

dhruvachak committed:
Updated OMPT tests that check for trace buffer flush. (#1576)

Account for 1 more flush because of recent change of handling of __omp_rtl_data_environment.

gregrodgers committed:
[srock] First commit of new srock system that will eventually replace aomp.  See README in srock-bin for details about srock

dpalermo committed:
[aomp][trunk] Change LLVM_LIT_ARGS to -j 16 (#1580)

estewart08 committed:
[rocmlibs] - Updates for move to 7.0 sources (#1582)

estewart08 committed:
Various Cherry Picks from 22.0-0 branch to aomp-dev (#1583)

* [build] - Update rocprofiler-sdk cmake prefix path

We need to pass the local directory where prerequisites are
located so that aqlprofile can be found especially when
there is no ROCm installed.

* [build] - copy aqlprofile libs into aomp install directory

rocprofiler-sdk.so depends on aqlprofile libs.

* [build] - Clean up debian packaging script

Remove /usr/bin links to deprecated tools.
Add additional rocmlibs exclusions for aomp debian.

* [rocmlibs][nfc] - fix source_dir typo in cmake echo command

* [build] - Update rccl and rocprofiler-sdk scripts to copy needed files into install

rccl - the .info directory from rocm-core is need in the base install of AOMP
as rccl does a version check based on this file. When we update the rccl
branch in the future there is an option to pass a ROCM_VERSION in via cmake.

rocprofiler-sdk - enhance conditional check for aqlprofile/lib.

* [build] - Add rocprofiler-sdk to tarball exclusions

* [packaging] - fix repo_name/dir_name comparison

There were recent changes that made dir_name a full path. We
now need to get the basename.

estewart08 committed:
Add 22.0-1 manifest

dpalermo committed:
[smoke-fort-dev] Add chk tests for new flags/features (#1581)

dhruvachak committed:
Added tests for small block size functionality for generic SPMD kernels. (#1584)

estewart08 committed:
[examples] - Remove tsp binary in stdpar/tsp

estewart08 committed:
[build] - use clang as cmake compiler for rocdbgapi

CatherineMoore committed:
Add MIT License. (#1585)

ronlieb committed:
quick fix: cmake problem for CMAKE_PREFIX_PATH

dhruvachak committed:
Remove small blocksize option for generic SPMD programs since it is now the default. (#1586)

estewart08 committed:
[docs] - Updates for 22.0-1 release

estewart08 committed:
[build] - Remove roctracer/rocprofiler from 22.0-1 manifest

gregrodgers committed:
[srock] checkout amd-staging or specified branch with SROCK_COMPILER_BRANCH for spirv-llvm-translator

gregrodgers committed:
[tr_aomp] [srock] srock now replaces tr_aomp

gregrodgers committed:
[srock] Latest patches for amd-staging

gregrodgers committed:
[srock] Add hip, hip_runtime, and hipify to the default minimal build

gregrodgers committed:
[srock] update the amd-staging patch set.  It is getting smaller

gregrodgers committed:
[run_accel2023.sh] Fix path to pickup clang/flang from old convention for AOMP

gregrodgers committed:
[FLANG] remove NOWAIT on TARGET TEAMS DISTRIBUTE PARALLEL DO because it now shows not imlemented

gregrodgers committed:
[srock] fix amd-staging patch to work against current TheRock tip

gregrodgers committed:
[srock] improve start and end banners to show better information on the build including data prep and build time breakouts

dhruvachak committed:
Fixed iteam reduction test by replacing num_threads with thread_limit clause. (#1589)

After a recent change, generic SPMD kernels may have their blocksize
reduced by the runtime. So the user code can no longer assume that the
number of threads launched will match the default blocksize. The only
way to ensure that is to add a thread_limit clause to the target
construct.

dpalermo committed:
[srock] Update where .amd-llvm.smrev is regenerated (#1587)

- Move to after amd-llvm branch is checked out/updated

estewart08 committed:
[build] - Standalone asan openmp build needs an additional path to locate devicelibs (#1590)

gregrodgers committed:
[srock] fix patch for compiler/pre_hook_amd-llvm

gregrodgers committed:
[srock] update patch to work with compiler/CMakelists.txt

gregrodgers committed:
[srock] Only apply _TheRock.patch if it exists for SROCK_COMPILER_BRANCH

skatrak committed:
[smoke-fort] Update kernel-exec-modes test

This test only passed if parallel regions were executed by multiple threads.
This is too restrictive and causes the test to fail if executed in Generic
mode.

Instead of failing, this patch updates the test to print a warning message to
indicate that performance might be impacted when running in single-threaded
mode.

raghavendhra committed:
Update versions of build_supp components openmpi(to 5.0.8), hdf5(to 1.14.0) and silo(to 4.11.1) used in GenASiS testing.

estewart08 committed:
[build] - Bump version for 22.0-2 build

dpalermo committed:
[smoke-fort-dev] Add target-cmplx[48]-div (#1594)

ronlieb committed:
Move gpurun to llvm-project/offload/utils/gpurun (#1591)

jplehr committed:
[CK] Add targets for smoke and regression build targets (#1597)

This change enables the two build system targets 'smoke' and
'regression' from CK to be targeted with this script.

Check the CK README for details, in a nutshell:
- smoke runs shorter tests
- regression takes longer

estewart08 committed:
[build] - Move llvm-project, SPIRV, hipify repos to clone from github (#1596)

Preemptive patch that we can land once the move has been completed.

gregrodgers committed:
[srock] build offload-arch, amdgpu-arch, and nvptx-arch

dpalermo committed:
[UMT] Update llvm-flang location for openmpi build

Kewen12 committed:
[BuildBot] Bump rocm version in dockerfiles (#1612)

agozillon committed:
[Flang][smoke-test] Add smoke-fort-dev test for flang-561059

skatrak committed:
[smoke-fort-fails] Add tests for outlined parallel regions

This patch adds a few tests including various edge cases found while
implementing support for the execution of parallel regions located in functions
called from target regions.

skatrak committed:
[smoke-fort-fails] Add test for a parallel reduction in a generic kernel

This test triggers a known issue where certain generic kernels are
misidentified as SPMD, resulting in wrong results.

For this problem to be addressed, the execution of parallel regions in generic
target regions must be fixed and also the detection of the kernel type must be
updated to avoid detecting such target regions, holding sequential code, as
SPMD.

skatrak committed:
[smoke-fort-dev] Add test for variable size array arguments

This test, which already works when running in SPMD mode (current behavior)
triggers an edge case for Generic mode that causes results to be incorrect.

This is related to the fact that, in that execution mode, a temporary allocated
to store the upper bounds prior to entering the parallel region should use
device shared memory instead of stack memory.

carlobertolli committed:
Add test for affinity clause of task directive. (#386)

Obtained from OpenMP 5.2 examples.

DominikAdamski committed:
[smoke-fort-fails] Added test case for flang-535416 (#1634)

This test case shows that the problem is related to nvars.
Flang runtimes calls are eliminated by manual assignment
of arrays. The test passes only if we replace nvars
with corresponding constant in the target region.

estewart08 committed:
[build] - Move to ROCm 7.1 sources (#1637)

- Fixed 'clone_aomp.sh list' to not report error if .git is used to clone repo
- Updated build_rocBLAS.sh to include -DROCTX_PATH and replaced deprecated
  -DAMDGPU_TARGETS with -DGPU_TARGETS. ROCTX (roctracer) headers and libraries
  are built as a part of rocprofiler-sdk. Now the header is found in include/rocprofiler-sdk-roctx
  and the library is named librocprofiler-sdk-roctx.so.
- Updated various rocmlibs patches to remove gfx1103. 7.1 added support for gfx1103.

ronlieb committed:
gfx1200 memory limits milestone3-babel

swdev-565113

 Please enter the commit message for your changes. Lines starting

ronlieb committed:
[smoke] sete envvar to 512 for xtream-red (#1641)

dhruvachak committed:
Fixed smoke-dev clang-ifaces. (#1648)

By removing -O0 from the compile line. Replicated the failing -O0
test in smoke-fails.

ronlieb committed:
[smoke-dev] force blkSize 512 for a few xteam test (#1647)

ronlieb committed:
[testing] add run_miniapps_test.sh and clean up hpc2021 and npsdb_build_ompi (#1622)

* [testing] add run_miniapps.sh and clean up hpc2021 and npsdb_build_ompi

* cleanup script

dpalermo committed:
[build_aomp] Add support for -DLLVM_FORCE_VC_REVISION/REPOSITORY (#1656)

- Mechanism for adding +PATCHED when using build_aomp.sh/build_trunk.sh
    - Setting following env vars adds corresponding LLVM cmake -D option:
        LLVM_FORCE_VC_REVISION
        LLVM_FORCE_VC_REPOSITORY

dpalermo committed:
[build_supp] Use relative symlinks & honor AOMP_JOB_THREADS (#1657)

dpalermo committed:
[build_srock] Add extra information to -DLLVM_FORCE_VC_REVISION (#1658)

- Mechanism for adding +PATCHED when using build_srock.sh
    - Via the .amd-llvm.smrev mechanism currently used by TheRock
    - Setting following env var adds to -DLLVM_FORCE_VC_REVISION:
        LLVM_SHA_EXTRA

estewart08 committed:
[build] - Fix issue with hipamd using comgr from /opt/rocm (#1668)

Hipamd searches for comgr config version 2.9 first and then 3.0.
There is a case where an older ROCm install will satisfy
the 2.9 version. This leads to issues when searching for hsa headers due to adding
-isystem /opt/rocm/include. The hsa headers from the older ROCm do not
build with newer ROCclr sources. For now we will remove /opt/rocm
from the find_package search path.

dpalermo committed:
[build_supp] Have readlink canonicalize/follow relative symlinks (#1672)

- Needed for install check to find previous versions via relative links
    - From: "Use relative symlinks & honor AOMP_JOB_THREADS (#1657)"

dpalermo committed:
[srock] Update _TheRock.patch to avoid merge conflict (#1677)

ronlieb committed:
[run_miniapps_test.sh] bypass gpurun for 2 spec suites

estewart08 committed:
[build] - Support for almalinux release build

estewart08 committed:
Add 22.0-2 manifest

mhalk committed:
[CI][OpenMP_VV] Match HSA_XNACK value of STR to batch execution

Fix behavior of Single-Test-Run to match batch execution of tests

DominikAdamski committed:
[NFC][testing] Prepare tests for implicit Flang GPU runtime (#1705)

Add one switch which can control if we want to test AOMP Flang compiler with implicitly linked Flang GPU runtime.
Do not add -lflang_rt.hostdevice flag if IMPLICIT_GPU_FLANG_RT is set to 1.

nicebert committed:
Fixes OpenMP examples to correctly determine XNACK capable GPUs

Adds logic to find_gpu_and_install_dir to set new environment variable LLVM_GPU_IS_XNACKABLE.
Changes logic in demo_offload_types test to use variable instead of checking GPU for "sm_" string.

dhruvachak committed:
Fixed tests smoke-dev mpi-reduce and mpi-allreduce. (#1714)

Make sure offload-arch has the right value.

Pick up gpurun from the correct dir.

jplehr committed:
[omptests] Output git log info (#1719)

Show the current commit in the repo. This is mostly such that a log file
does contain the info which commit was used in the run.

nicebert committed:
[calculate_divergence.sh] adds script to easily calculate divergence

Script to calculates and report the divergence between LLVM upstream and ROCm's LLVM
fork by comparing git branches and generating diff statistics and patches.
Runs out of the box using some default configurations doing required setup for
directories, remotes, ...
Includes documentation.

ronlieb committed:
Add zach and carlo as codeowners

ergawy committed:
[smoke-fort][OpenMP] Add tests for GPU reductions on allocatables and static arrays (#1718)

* [smoke-fort][OpenMP] Add tests for GPU reductions on allocatables and static arrays

Adds smoke tests for the changes introduced in upstream PR: https://github.com/llvm/llvm-project/pull/165714. These tests cannot be added as offloading tests upstream (yet) since they need to link wit h flang's device RT lib.

* review comments, Michael

jplehr committed:
[CK] Use make as fallback when ninja is not avail (#1727)

This can happen on some machines on which we have limited control what
software is installed.

ronlieb committed:
[palameda] fix copyrights https://ontrack-internal.amd.com/browse/SWDEV-569327 (#1726)

Co-authored-by: CatherineMoore <catmoore@amd.com>

jplehr committed:
[CK] Fix CMake generator string (#1732)

In https://github.com/ROCm/aomp/pull/1727 I did not get the escaping of
\" right. This PR will simply use CMake defaul (Unix Makefiles) when
ninja is not present and explicitly specify ninja when it is available.

nicebert committed:
[calculate_divergence.sh] removes default component "driver"

Removes default component "driver" accidentally commited from testing.
Allows the script to default run on entiry llvm-project directory.

jplehr committed:
[CK] Replace hardcoded 'ninja' with $CKBuildTool (#1734)

Another follow-up on https://github.com/ROCm/aomp/pull/1732 to not rely
on the hardcoded 'ninja' invocation but adjust the tool that is used
depending on whether ninja is available.

nicebert committed:
[calculate_divergence.sh] adds functionality to create file (group) based diffs

Saves created diff files into timestamped subdirectory.

dpalermo committed:
[build_supp] Update to ninja 1.13.2 (#1740)

Environment variable support:
    - NINJA_NPROCS  - specify number of procs to use
    - NINJA_STATUS  - show detailed build progress
    - NINJA_VERBOSE - show command lines while building

dpalermo committed:
[srock] Build ninja, define SROCK_JOB_THREADS & NINJA_NPROCS (#1742)

* [srock] Build ninja, define SROCK_JOB_THREADS & NINJA_NPROCS

* [srock] Define -DTHEROCK_BACKGROUND_BUILD_JOBS=1

    - Process one component at a time
    - Provides better control of total job parallelism

dpalermo committed:
[srock] Enable LLVM_OFFLOAD_BINARY & LIT testing (#1743)

    - Update tools for +LLVM_OFFLOAD_BINARY / -CLANG_OFFLOAD_PACKAGER
    - Enable all tools
    - Enable LIT testing

dpalermo committed:
[srock] Allow building flang-rt in TheRock amd-llvm (#1744)

    export AOMP_REPOS="${SROCK_REPOS}/TheRock/compiler"
    export BUILD_AOMP="${SROCK_REPOS}/TheRock"
    export BUILD_AOMP_SUBDIR="build/compiler/amd-llvm"
    export AOMP_NAME_LLVM_PROJECT="amd-llvm"
    ./build_llvm-flang-rt-host-dev.sh

estewart08 committed:
Run rocm test therock support (#1749)

* run_rocm_test - Initial changes to support TheRock testing

* run_rocm_test - Update rocmver detection

Properly initialize rocmver and therock variables.

* run_rocm_test - Guard rocmver detection

* run_rocm_test - Update ROCMINF/ROCMDIR path

- Add path to usage of rocminfo/rocmsmi

* Makefile.defs - Update FileCheck Search

Do not look in openmp-extras package anymore. Instead look
in HOME directory.

* run_rocm_test - Download gpurun and filecheck locally if missing in install

* [smoke] - Add rpath to clang-392854, libgomp-292348

* Remove newline deletion in run_nekbone.sh

Lynd98 committed:
Changed to run_RSBench.sh, run_XSBench.sh, and run_su3bench.sh to get them working in CI.
Added these benchmarks to the manifest, they were missing.

jplehr committed:
[su3bench] Fix remote in manifest file for su3_bench (#1766)

nicebert committed:
[calculate_divergence.sh] adds improvements to script

Refactoring of git --merge-base command into function for easier reuse.
Adds functionality to calculate "non-functional" changes patch for
whitespaces, braces, ...
Creates ".commits" file listing commit information that introduced divergence
with information about the files changed and the lines of difference.
Improves logged information to include script environment.

ronlieb committed:
Add api test for flang-56801 (#1769)

* Add api test for flang-56801

* Add test from user

DominikAdamski committed:
[smoke-fort-fails] Added test case for flang-535416 (#1770)

It's legal OpenMP kernel and it should execute without
segfault.

dpalermo committed:
[srock] Add GFXFAM to specify -DTHEROCK_AMDGPU_FAMILIES (#1773)

- Needed for subproject therock-boost

estewart08 committed:
[run_hpc2021] - Add UCX_LOG_LEVEL=error

This seems to have been done in amd-staging only and is causing merge issues
with 'git am'. Applying it to aomp-dev to later flow back into amd-staging.

estewart08 committed:
Remove extra whitespace in calculate_divergence.sh

'git am' is complaining about whitespace errors.

estewart08 committed:
[test] - Update tests that need SUPPORTS_USM remove check_smoke_dev.sh (#1772)

- check_smoke_dev.sh now symlinks to ../smoke/check_smoke.sh like other directories do.
- Add  to smoke/usm-globals and smoke-limbo/usm-locals-xnack-enabled-xnack-minus/Makefile. These
tests should not run on a system that does not support USM.

jplehr committed:
[su3bench] Fix directory name (#1786)

With the update to the manifest I landed, I broke this script. Update
the script to account for new location name.

amitamd7 committed:
offloading_disabled_resolve_smoke_fort_test (#1787)

DominikAdamski committed:
[Flang-RT] No flang hostdevice runtime flag (#1756)

* [Flang-RT] No flang hostdevice runtime flag

* Use new Flang GPU runtime by default

ronlieb committed:
bump mod from 2 to 3 for aomp (#1798)

dpalermo committed:
[srock] Updates for building amdgcn-amd-amdhsa/libflang_rt.runtime.a (#1803)

ronlieb committed:
mv  flang-274768-1 ../smoke-fort-dev/ (#1810)

dpalermo committed:
[smoke-fort-fails] Add flang-527182

ronlieb committed:
Add gpurun tests

e-kwsm committed:
chore: remove vim swap files (#1386)

Co-authored-by: Eisuke Kawashima <e-kwsm@users.noreply.github.com>

ronlieb committed:
Remove classic flang repos from manifests 22 onward (#1819)

ronlieb committed:
[babelsteam] use bin/clang instead of bin/llc for --version (#1825)

ronlieb committed:
[smoke-fort-{dev,limbo}] test move (#1828)

* [smoke-fort-{dev,limbo}] test move

* gfx1030 needs flang-519158 to move back to smoke-fort-dev

ronlieb committed:
[aomp/test/] move passing smoke-dev to smoke-limbo (#1829)

* [aomp/test/] move passing smoke-dev to smoke-limbo

* [smoke] move some tests back to dev mi350,gfx1030

ronlieb committed:
[smoke-dev] Move 12 tests back here from limbo (#1832)

ronlieb committed:
[smoke-fort-dev] test case swdev-566120 (#1834)

ronlieb committed:
[smoke-fort-dev] add test for llvn swdev-566304 (#1831)

ddpagan committed:
[smoke-limbo] multi-sdma not supported on gfx1151 (#1836)

ddpagan committed:
[smoke-limbo] aomp_mappings_newrt should account for number of teams (#1840)

For the 'target teams' case, the number of teams can exceed the size of
the arrays used to gather information (for example, on gfx942), which can
cause the test to fail. To ensure array assignments are always inbounds,
and to be consistent with the other tests, limit assignments to only those
array elements actually checked after the 'target teams' region.

jplehr committed:
[CK] Allow build only (#1843)

Previously some workload was always run. This change allows to only
build the CK library.

This is to split the nightly jobs further and try to contain build
issues to smaller areas.

dpalermo committed:
[umt] Default to IMPLICIT_GPU_FLANG_RT=1 (#1848)

dpalermo committed:
[srock] Update _TheRock.patch to avoid merge conflict (#1849)

estewart08 committed:
Adjust aomp version back to 22.0-2 after rebase with aomp-dev

estewart08 committed:
[build] - Bump major version to 23 (#1856)

* [build] - Bump major version to 23

* [srock] - update common vars to 23.0-0

estewart08 committed:
Update 22.0-2 manifest to use 7.1.1 sources

mhalk committed:
[CK] Add flag to enable library-only build

Add '-l' flag to omit tests and examples during CK library build

dpalermo committed:
[srock] Update _TheRock.patch to avoid merge conflict (#1858)

estewart08 committed:
[rocmlibs] - Update tensile patch to fix v_dot4 op_sel modifiers

estewart08 committed:
[docs] - Update for 22.0-2 release

estewart08 committed:
[build] - Add LIBC_INCLUDE_BENCHMARKS=0 to tarball creation

There are some benchmarks that use cmake functions that
are not reachable when DISABLE_LLVM_TESTS=1.

ronlieb committed:
[smoke] remove veccopy-ompt-target-emi-tracing (#1865)

ronlieb committed:
[smoke-limbo] make non blocking (#1868)

jplehr committed:
[CK] Enable user-defined build paralellism (#1873)

ronlieb committed:
Revert "[smoke-limbo] make non blocking (#1868)" (#1874)

This reverts commit bbc77cfb569dc13320f45aadd5630107265297b7.

estewart08 committed:
[build] - Updates for move to ROCm 7.2 sources (#1871)

mjklemm committed:
Add test flang-577054

ronlieb committed:
[theRockCI] add script run_theRockCI.sh, use python filecheck (#1882)

* [theRockCI] add script run_theRockCI.sh, use python filecheck

* Add scrupt run_theRockCI.sh

jplehr committed:
[LLaMA] Add run script for LLaMA C++ (#1867)

* [LLaMA] Add run scritp for LLaMA

To enable running this in our nightlys, add a simple script to clone /
update and configure / build LLaMA.
Also allows running their test via ctest.

* Fixing missing num parallel jobs

ronlieb committed:
Remove lines 109,110 from test/Makefile.defs emergency fix

dpalermo committed:
[srock] Update _TheRock.patch to avoid merge conflict (#1889)

dpalermo committed:
[srock] Add SROCK_THEROCK_BRANCH=compiler/amd-staging (#1899)

- Leverages TheRock changes made by the CI team for amd-staging
   - Fixes error seen with hip-tests

ronlieb committed:
[run_theRock] pass/fail fixes, timing expectations (#1905)

added SKIP_QUICK and SKIP_CLONE

ronlieb committed:
[smoke-limbo] mv to test to smoke-dev: veccopy-ompt-target-tracing-flush-only-on-api-use veccopy-ompt-target-tracing-flush-only-on-shutdown (#1906)

ro-i committed:
Fix veccopy-ompt-target-tracing-flush-only-on-api-use/-on-shutdown

ro-i committed:
move tests back to smoke-limbo

jplehr committed:
[CK] Build library targets first (#1939)

Building CK takes long, so we want to get as much done as possible.
To do that, we want to keep going to catch as many errors as possible.
That is why `-k` is added to the build command.

We also want to restrict it to only the library components first. To
that end, we build ckProfiler which appears to depend on the library
components and enables us to do that without restricting the CMake
configuration to only the library parts.

That means in a subsequent step we can still build smoke / regression
tests etc.

ronlieb committed:
make hpc2021 nonblock, and ulimit -t 200 (#1941)

jplehr committed:
[LLaMA] Update ctest log file location (#1944)

This updates the ctest log file location to work better with the
internal nightly automation

jplehr committed:
[LLaMA][NFC] Update formatting via shfmt (#1945)

jplehr committed:
[CK] Pull source from rocm-libraries super repo (#1943)

CK moved into the rocm-libraries super repo as a project. Update URLs
and how it is cloned / pulled and configured.

jplehr committed:
[CK][NFC] Reformat run script with shfmt (#1946)

jplehr committed:
[CK][LLaMA] Add license headers to run scripts (#1947)

jplehr committed:
[CK] Simplify grep for build success (#1948)

jplehr committed:
[LLaMA] Add LLaMA C++ benchmark (#1950)

We also want to run performance benchmarks. This adds the initial
infrastructure to do so.
It uses a randomly picked "trending" model from hugging face that is not
too large to download.

jplehr committed:
[LLaMA] Use make if ninja not avail (#1949)

It seems that some of the test environments do not have the ninja-build
tool installed. Auto-detect these cases and use make instead.

jplehr committed:
[LLaMA] Feature flag to update src (#1951)

Allow to stay on a specific source version.

jplehr committed:
[LLaMA] Add git-lfs check for model download (#1952)

Some tests appears to download small models from huggingface using
git-lfs. Check for availability and give warning about potentially
failing tests.

estewart08 committed:
[build] - Bump version to 23.0-1

estewart08 committed:
[build] - Add manifest for 23.0-0 release

estewart08 committed:
[build] - Add new script to create release binary tarball

ronlieb committed:
[run_theRock] handle no ROCm variant (#1953)

ronlieb committed:
[run_theRockCI] cleanup to get to green (#1954)

ronlieb committed:
[run_theRock] fix Red Green (#1958)

estewart08 committed:
[build] - Pass python version to rocprofiler-sdk

There was an issue seen with alma8 where rocprofiler-sdk was picking
up the newest python version in /usr/local/bin, which is causing cmake
errors. Determine the current python3 default version and pass it through.

ronlieb committed:
reblock hpc2021 in run_rocm_test.sh (#1959)

jplehr committed:
[LLaMA] Always init git-lfs (#1962)

As noted in
https://github.com/ROCm/aomp/pull/1952#discussion_r2798595032 git-lfs
may require the use of `install` to make it functioning.

When available do that and pull-in assets.

dpalermo committed:
[srock] Update _TheRock.patch to avoid merge conflict (#1969)

ronlieb committed:
eliminate standalone build (#1975)

* eliminate standalone build

* Add new build_standalone

ronlieb committed:
[run_babelstream.sh] peg sha to 2f00dfb , patch apply issue

mhalk committed:
[CK] Add installation success marker and ignore intermediate errors

Add -k 0 argument to installation
 - CK installation may fail intermediately, hiding other "build" errors

Add print of CK-INSTALL-SUCCESS

estewart08 committed:
[docs] - Updates for 23.0-0

Deprecated docs for SLES, Ubuntu, and RHEL/CentOS installs. We now build
on AlmaLinux and distrubute a tar file that can be used on many operating
systems.

estewart08 committed:
[docs] - Additional updates for AlmaLinux build instructions

estewart08 committed:
[docs] - deprecate spack install instructions

estewart08 committed:
[docs] - deprecate additional spack install instructions

estewart08 committed:
Update build_aomp_docker_release.sh for 23.0-0

dpalermo committed:
[UMT] Reduce OMP_DEVICE_TEAM_THREAD_LIMIT 1024 to 512 (#1978)

- Avoids register allocation issue at runtime
   - Which is incorrectly reported by HSA as:
     HSA_STATUS_ERROR_INVALID_ISA: The instruction set architecture is invalid.

Co-authored-by: Abid Qadeer <haqadeer@amd.com>

dpalermo committed:
[UMT] Add settings for RX7900XT*/gfx1100 (#1979)

ronlieb committed:
more cleanup for run_theRock.sh (#1980)

ronlieb committed:
[run_theRock.sh] add more filecheck support

ronlieb committed:
[run_theRock.sh] more filecheck foo

ronlieb committed:
[run_theRock.sh] more filecheck and gpurun for theRock tests

ronlieb committed:
[run_theRock.sh] mechansim to hide unrunnable tests

ronlieb committed:
[run_theRock.sh] hide flang-gpu-abort from smoke-fort

ronlieb committed:
[run_theRock.sh] exit 0 if pass

ronlieb committed:
[run_theRock.sh] hide usm-globals-with-pragma from smoke

ronlieb committed:
[run_theRock.sh] hide more usm tests

ronlieb committed:
[run_theRock.sh] remove dup flang-gpu-abort

ronlieb committed:
[run_theRock.sh] update S/SF times

jplehr committed:
Address review comment

jplehr commented:
Apologies for the rebase f-up.

jplehr commented:
Opened https://github.com/ROCm/aomp/pull/1989 as clean version.
